### PR TITLE
feat: add storage API

### PR DESCRIPTION
**Pull request by andrewrynhard**
This is the initial implementation of a storage API.

Signed-off-by: Andrew Rynhard <andrew@rynhard.io>

### CODE REVIEW

File: internal/app/maintenance/server/server.go [hunk 1]
```diff
@@ -61,3 +64,27 @@ func (s *Server) ApplyConfiguration(ctx context.Context, in *machine.ApplyConfig
 
 	return reply, nil
 }
+
+// Disks implements machine.MaintenanceService.
+func (s *Server) Disks(ctx context.Context, in *empty.Empty) (reply *storage.DisksResponse, err error) {
```
Comment (smira) [inline internal/app/maintenance/server/server.go:69]: we need to add a registration line to the `Register` method

Comment (andrewrynhard) [reply internal/app/maintenance/server/server.go:69]: ah, right :+1:

**Review by Unix4ever — DISMISSED** [review]

**Review by smira — APPROVED** [review]

**Pull request merged**